### ISSUE TITLE
sudo doesn't work on non-remote connections

### ISSUE THREAD

**Issue by KDr2** [RexOps/Rex#613]
The target machine is a debian linux.
My simple task:

```
task test => sub {
   sudo sub{ say run "mkdir /rex-test"; }
};
```

When I run `rex -dd test` on a linux machine, it fails, the log: http://tny.cz/dde13ad6
But when I run it on MacOSX, it works well, the log: http://tny.cz/ceb8bbc2
I tested this with perl5.16 and perl5.20

**Comment by krimdomu** [comment]
Hi KDr2,

thanks for the report. As a workaround you can connect to your linux system via ssh to localhost. (rex -H localhost test).

This problem seems to be happening when rex doesn't do ssh connections.

**Comment by KDr2** [comment]
Thanks,  `rex -H localhost test` works.

**Comment by AlexanderS** [comment]
Any fix for that (apart from the workaround)?

**Comment by KDr2** [comment]
Also want to know if there's any fix for that (apart from the workaround)?

**Comment by ferki** [comment]
Hmm..it seems it has been already fixed in 1.5.0, with [this commit](https://github.com/RexOps/Rex/commit/40995ae2d92ebac6dd7dcf39f5ba110cfc6f0f59). The following `Rexfile` seems to work properly for me with the latest commit from `development-1.x` branch:

```perl
use Rex -feature => [ '1.4', 'exec_autodie' ];

sudo TRUE;
sudo_password 'mysudopassword';

task 'localtest', sub {
    say run 'whoami';
};
```

```bash
$ rex localtest
[2017-11-30 15:01:26] INFO - Running task localtest on <local>
root
[2017-11-30 15:01:26] INFO - All tasks successful on all hosts
```

@AlexanderS @KDr2: could you check which version of Rex you have installed, please, and also try the above test task with on your machine? Let us know if it still fails for your case!

**Comment by KDr2** [comment]
I just upgrade to 1.5.0, and Rex works well with the Rexfile you gave. The fix is confirmed, thanks.
